### PR TITLE
Refactor test_api_endtoend and endtoend/utils for pytest

### PR DESCRIPTION
**Pull request by mshriver**
Running these locally still, not quite ready yet.

`tests/foreman/endtoend/test_api_endtoend.py::AvailableURLsTestCase::test_positive_get_links` Is failing in the master branch, I won't be addressing that failure here.  Some API endpoints have changed in recent snaps.

```
setup@localhost:~/repos/robottelo (pytest-api-endtoend-13994$%=) % pytest -v tests/foreman/endtoend/test_api_endtoend.py 
======================================================================================== test session starts =========================================================================================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/setup/repos/robottelo/.robottelo/bin/python3.8
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/setup/repos/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, ibutsu-1.14.1, xdist-2.2.1, mock-3.5.1, reportportal-5.0.8, cov-2.11.1, services-2.2.1
collected 7 items                                                                                                                                                                                    

tests/foreman/endtoend/test_api_endtoend.py::TestAvailableURLs::test_positive_get_status_code PASSED                                                                                           [ 14%]
tests/foreman/endtoend/test_api_endtoend.py::TestAvailableURLs::test_positive_get_links FAILED                                                                                                 [ 28%]
tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_find_default_org PASSED                                                                                               [ 42%]
tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_find_default_loc PASSED                                                                                               [ 57%]
tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_find_admin_user PASSED                                                                                                [ 71%]
tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_ping SKIPPED (BZ:1897360)                                                                                             [ 85%]
tests/foreman/endtoend/test_api_endtoend.py::TestEndToEnd::test_positive_end_to_end PASSED                                                                                                     [100%]

============================================================================================== FAILURES ==============================================================================================
<KNOWN FAILURE ON MASTER BRANCH>
====================================================================================== short test summary info =======================================================================================
FAILED tests/foreman/endtoend/test_api_endtoend.py::TestAvailableURLs::test_positive_get_links - AssertionError: API path mismatch (expected first): 
================================================================== 1 failed, 5 passed, 1 skipped, 74 warnings in 126.76s (0:02:06) ===================================================================

```

### CODE REVIEW

File: tests/foreman/endtoend/test_api_endtoend.py [hunk 1]
```diff
@@ -1204,9 +1191,9 @@ def test_positive_end_to_end(self):
             organization=org,
         ).create()
         # check that content view matches what we passed
-        self.assertEqual(content_host.content_facet_attributes['content_view_id'], content_view.id)
+        assert content_host.content_facet_attributes['content_view_id'], content_view.id
         # check that lifecycle environment matches
-        self.assertEqual(content_host.content_facet_attributes['lifecycle_environment_id'], le1.id)
+        assert content_host.content_facet_attributes['lifecycle_environment_id'], le1.id
```
Comment (JacobCallahan) [inline tests/foreman/endtoend/test_api_endtoend.py:1196]: This assertion should be ==

File: tests/foreman/endtoend/test_api_endtoend.py [hunk 2]
```diff
@@ -1204,9 +1191,9 @@ def test_positive_end_to_end(self):
             organization=org,
         ).create()
         # check that content view matches what we passed
-        self.assertEqual(content_host.content_facet_attributes['content_view_id'], content_view.id)
+        assert content_host.content_facet_attributes['content_view_id'], content_view.id
```
Comment (JacobCallahan) [inline tests/foreman/endtoend/test_api_endtoend.py:1194]: This assertion should be ==

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/endtoend/test_api_endtoend.py (hunk 2 above)
Comment (mshriver) [reply tests/foreman/endtoend/test_api_endtoend.py:1194]: Good catches, thanks!

**Comment by mshriver** [comment]
trigger: test-robottelo
pytest: tests/foreman/endtoend/test_api_endtoend.py

**Comment by JacobCallahan** [comment]
Looks like you had a failure in the prt

**Comment by mshriver** [comment]
@JacobCallahan yep, check out the description - its a failure on the master branch, some API endpoints have changed in recent snaps.

**Review by JacobCallahan — APPROVED** [review]

**Review by rplevka — APPROVED** [review]

**Pull request merged**